rust/lance/src/index/vector/builder.rs

```diff
@@ -271,31 +278,33 @@ impl<S: IvfSubIndex, Q: Quantization + Clone> IvfIndexBuilder<S, Q> {
                 let reader =
                     FileReader::try_open(scheduler.open_file(&index_part_path).await?, None)
                         .await?;
-                let batch = reader
-                    .read_stream(ReadBatchParams::RangeFull, index_size as u32, 1)?
-                    .try_next()
-                    .await?
-                    .ok_or(Error::io("empty index batch", location!()))?;
+                let batches = reader
+                    .read_stream(ReadBatchParams::RangeFull, storage_size as u32, 1)?
+                    .peekable()
+                    .try_collect::<Vec<_>>()
+                    .await?;
+                let batch = arrow::compute::concat_batches(&batches[0].schema(), batches.iter())?;
                 index_writer.write_batch(&batch).await?;
+                index_ivf.add_partition(batch.num_rows() as u32);
             }
         }
 
         let mut storage_writer = storage_writer.unwrap();
         let storage_ivf_pb = pb::Ivf::try_from(&storage_ivf)?;
+        storage_writer.add_schema_metadata(DISTANCE_TYPE_KEY, self.distance_type.to_string());
         storage_writer.add_schema_metadata(
             IVF_METADATA_KEY,
-            String::from_utf8(storage_ivf_pb.encode_to_vec()).map_err(|e| {
-                Error::io(format!("failed to encode IVF metadata: {}", e), location!())
-            })?,
+            hex::encode(storage_ivf_pb.encode_to_vec()),
```

Comment:
why do we use hex encoding here? can we just write some random data? We do have too many dependencies now.

Reply:
we need hex for converting the bytes to string cause the schema metadata can accept `String` only, any recommendations here?